packages/react-dom/src/__tests__/ReactTreeTraversal-test.js

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+var React;
+var ReactDOM;
+
+var ARG = {arg: true};
+var ARG2 = {arg2: true};
+
+const ChildComponent = ({id, eventHandler}) => (
+  <div
+    id={id + '__DIV'}
+    onClickCapture={e =>
+      eventHandler(e.currentTarget.id, 'captured', e.type, ARG)
```

Comment:
The logic behind choosing to put either `ARG` or `ARG2` here is not clear to me.

In the original tests, it was specific to a test. We're traversing enter/leave, and calling "from" path with `ARG` and "to" path with `ARG2`.

Here, we instead encoded `ARG` and `ARG2` right into components. I don't think this is right. This seems to couple them with specific test cases below. So it might be hard or impossible to add other completely valid tests which test the opposite enter/leave paths. This didn't use to be a problem.

I'm not sure what the solution is; just pointing out an issue I see.

Reply:
Yeah I'm not entirely happy with that either. Will think about a more elegant solution.

Reply:
I wonder if we're good to just remove the `ARG`s since we're now tracking the event type which we haven't done before. If I get it right`traverseEnterLeave(from, to, fn, argFrom, argTo)` will be called with `argFrom` and `argTo` set to the events that are dispatched on the target anyway so I feel like extracting the target id and event type will be strict enough.

Reply:
Yes this sounds right